api/src/main/java/com/datatorrent/api/StatsListener.java

```diff
@@ -115,6 +115,28 @@
     List<OperatorResponse> getOperatorResponse();
   }
 
+  /**
+   * An interface to the DAG. Stats listener can get information about
+   * operator or other elements in the DAG through this interface. currerntly
+   * we only provide method to extract the operator name based on the physical
+   * id of the operator. In future more methods can be added.
+   *
+   */
```

Comment:
exposing all information may expose internal implementation of the DAG. we can only have dependency on apex-api. Can you suggest which information can be provided to stat listener without adding dependency on the apex-engine. I can think of following
- name
- number of partitions for self.
- aggregated stats at logical operator level.
This list could be expanded in future.

This jira puts the required infrastructure without breaking backward compatibility.